aws/resource_aws_elasticache_cluster.go

```diff
@@ -210,6 +217,81 @@ func resourceAwsElasticacheCluster() *schema.Resource {
 		},
 
 		Schema: resourceSchema,
+
+		CustomizeDiff: customdiff.Sequence(
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				// Plan time validation for az_mode
+				// InvalidParameterCombination: Must specify at least two cache nodes in order to specify AZ Mode of 'cross-az'.
+				azMode, azModeOk := diff.GetOk("az_mode")
+				numCacheNodes, numCacheNodesOk := diff.GetOk("num_cache_nodes")
+				if azModeOk && numCacheNodesOk && azMode.(string) == elasticache.AZModeCrossAz && numCacheNodes.(int) == 1 {
+					return errors.New(`az_mode "cross-az" is not supported with num_cache_nodes = 1`)
+				}
+				return nil
+			},
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				// Plan time validation for engine_version
+				// InvalidParameterCombination: Cannot modify memcached from 1.4.33 to 1.4.24
+				// InvalidParameterCombination: Cannot modify redis from 3.2.6 to 3.2.4
+				if diff.Id() != "" && diff.HasChange("engine_version") {
```

Comment:
Similar to the other short circuiting one, this flattens out if you just do the easy return first:

```golang
if diff.Id() == "" || !diff.HasChange("engine_version") {
  return nil
}

// rest of it, but flatter indentation
```